agent.py

```diff
@@ -75,7 +78,9 @@ def __init__(
         self,
         swarm_agent_config: SwarmAgentConfig,
         cluster_agent_config: ClusterAgentConfig,
+        swarm_client: SwarmApi,
```

Comment:
Pass this as part of `swarm_agent_config: SwarmAgentConfig`

Reply:
It is not a configuration

Reply:
`ClusterAgentConfig` is meant for cluster-specific parameters that the agent needs, `SwarmAgentConfig` is meant for generic utilities / parameters the agent needs (see it has kube_cache, executor, etc). The client object fits perfectly in there

Reply:
See also https://github.com/openshift-assisted/assisted-swarm/pull/2/files#r862448691

Reply:
Can do that, but I still think that in this case there is a difference between configuration class and functional class

Reply:
I agree that Config is not the best name, but that class was meant to give the agent all the dependencies it needs to function, including functional things (like KubeCache and the executor).